tsec-libsodium/src/main/scala/tsec/libsodium/cipher/cipher.scala

```diff
@@ -0,0 +1,111 @@
+package tsec.libsodium
+
+import cats.evidence.Is
+import tsec.common._
+
+package object cipher {
+
+  private[tsec] val Plaintext$$ : TaggedByteArray = new TaggedByteArray {
+    type I = Array[Byte]
+    val is = Is.refl[I]
+  }
+
+  type PlainText = Plaintext$$.I
+
+  object PlainText {
+    def apply[A](bytes: Array[Byte]): PlainText = is.flip.coerce(bytes)
+    @inline def is: Is[PlainText, Array[Byte]]  = Plaintext$$.is
+  }
+
+  private[tsec] val SodiumKey$$ : HKByteArrayNewt = new HKByteArrayNewt {
+    type Repr[A] = Array[Byte]
+
+    def is[G] = Is.refl[Array[Byte]]
+  }
+
+  /** Our newtype over private keys **/
+  type SodiumKey[A] = SodiumKey$$.Repr[A]
+
+  object SodiumKey {
+    def apply[A](bytes: Array[Byte]): SodiumKey[A]   = is[A].coerce(bytes)
+    @inline def is[A]: Is[Array[Byte], SodiumKey[A]] = SodiumKey$$.is[A]
+  }
+
+  private[tsec] val RawCiphertext$$ : HKByteArrayNewt = new HKByteArrayNewt {
+    type Repr[A] = Array[Byte]
+
+    def is[G] = Is.refl[Array[Byte]]
+  }
+
+  /** Our newtype over private keys **/
```

Comment:
very copy, much pasta.

I'd love to be able to scrap this by the end of the week.

Reply:
spaghettini